plugins/media-keys/gsd-media-keys-manager.c

```diff
@@ -941,6 +946,11 @@ do_touchpad_action (GsdMediaKeysManager *manager)
         GSettings *settings;
         gboolean state;
 
+        if (manager->priv->lid_is_closed) {
```

Comment:
Instead of saving states, you could just query for `up_client_get_lid_is_present() && up_client_get_lid_is_closed()` from the `UpClient` here.
That would remove the need to connect to the state notifications.